src/controllers/auth.controller.js

```diff
@@ -26,23 +26,23 @@ const authController = (User) => {
     }));
   };
 
-  const register =(req, res) => {
-      var user = new User();
-       user.set("name",req.body.name);
-       user.set("email", req.body.email);
-       user.set("password", req.body.password);
-
-       user.save().then((user)=> {
-          res.status(200).json({message:'User created successfully with name: ' + user.get("name") + 'and email: ' + user.get("email")});
-      }).catch((err)=>{
-        res.status(400).json({
-          error: err,
+  const register = (req, res) => {
+    const { role, name, password, email } = req.body
+    User.create({
+      role, name, email, password
+    }).then((user) => {
+      res.status(200).json({ message: 'User created successfully with name: ' + user.get("name") + ' and email: ' + user.get("email") });
```

Comment:
perfect! just change user.get('name') here to just user.name and user.email since we have a user object in our response